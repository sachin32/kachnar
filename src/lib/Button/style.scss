.k_button {
  background-color: #3C89E8;
  border: 1px solid #7B9DF2;
  border-radius: 3px;
  font-size: 16px;
  padding: 3px 12px;
  color: #FFFFFF;

  &--primary {
    background-color: #3C89E8;
    border-color: #7B9DF2;
  }

  &--success {
    background-color: #2FB948;
    border-color: #43C66D;
  }

  &--error {
    background-color: #EC5858;
    border-color: #A25151;
  }

  &--warn {
    background-color: #EEBC69;
    border-color: #BFB15D;
  }

  &--info {
    background-color: #7CB7E8;
    border: 1px solid #64A0D2;
  }

  &--micro {
    font-size: 10px;
  }

  &--mini {
    font-size: 12px;
  }

  &--small {
    font-size: 14px;
  }

  &--big {
    font-size: 16px;
  }

  &--xbig {
    font-size: 18px;
  }
}