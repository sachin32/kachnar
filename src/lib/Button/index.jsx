import React from 'react';
import PropTypes from 'prop-types';
import _omit from 'lodash/omit';
import BEM from '../../utils/bem';

import './style.scss';

const Button = props => {
    const { size, type, children, className, onClick, ...restProps } = _omit(
        props, ['primary', 'success', 'error', 'warn', 'info', 'mini', 'micro', 'small', 'big', 'xbig']
    );
    const classes = BEM('k_button')({
        primary: props.primary,
        success: props.success,
        error: props.error,
        warn: props.warn,
        info: props.info,
        mini: props.mini,
        micro: props.micro,
        small: props.small,
        big: props.big,
        xbig: props.xbig,
    });
    return (
        <button
            className={classes.mix(className)}
            type={type}
            onClick={onClick}
            {...restProps}>
            {children}
        </button>
    );
};

Button.propTypes = {
    primary: PropTypes.bool,
    success: PropTypes.bool,
    error: PropTypes.bool,
    warn: PropTypes.bool,
    info: PropTypes.bool,
    mini: PropTypes.bool,
    micro: PropTypes.bool,
    small: PropTypes.bool,
    big: PropTypes.bool,
    xbig: PropTypes.bool,
    size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    type: PropTypes.string,
    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.number]).isRequired,
    className: PropTypes.string,
};

Button.defaultProps = {
    primary: false,
    success: false,
    error: false,
    warn: false,
    info: false,
    mini: false,
    micro: false,
    small: false,
    big: false,
    xbig: false,
    size: null,
    type: 'button',
    className: null,
};

export default Button;