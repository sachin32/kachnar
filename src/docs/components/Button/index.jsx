import React from 'react';
import { Button } from "../../../lib";

const styleObj = {
    marginRight: '5px'
};

const ButtonExample = props => {
    return (
        <div style={{ margin: '10px' }}>
            <h2>Button</h2>
            <Button style={styleObj}>
                Default
            </Button>
            <Button success style={styleObj}>
                Success
            </Button>
            <Button error style={styleObj}>
                Error
            </Button>
            <Button warn style={styleObj}>
                Warn
            </Button>
            <Button info style={styleObj}>
                Info
            </Button>

            <h2>Button Size</h2>
            <Button style={styleObj} micro>
                Micro
            </Button>
            <Button success style={styleObj} mini>
                Mini
            </Button>
            <Button error style={styleObj} small>
                Small
            </Button>
            <Button warn style={styleObj} big>
                Big
            </Button>
            <Button info style={styleObj} xbig>
                Extra Big
            </Button>
        </div>
    );
};

export default ButtonExample;