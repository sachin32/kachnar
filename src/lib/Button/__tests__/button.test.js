import React from 'react';
import { shallow } from 'enzyme';
import Button from "..";

describe('Button', () => {
    it('should render correctly in primary mode', () => {
        const component = shallow(<Button>Primary</Button>);
        expect(component).toMatchSnapshot();
    });
});