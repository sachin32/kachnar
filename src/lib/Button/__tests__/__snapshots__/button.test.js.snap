// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly in primary mode 1`] = `
<button
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  type="button"
>
  Primary
</button>
`;
